platform/migration/platform-migratable-util/src/main/java/org/codice/ddf/migration/util/MigratableUtil.java

```diff
@@ -0,0 +1,275 @@
+     * must not contain any symbolic link, otherwise the file will not be copied and a
+     * {@link MigrationWarning} will be returned.
+     *
+     * @param propertyFilePath path to the Java properties file that contains the path to the
+     *                         source file to copy
+     * @param javaProperty     name of the property inside the Java properties file that contains
+     *                         the path to the source file
+     * @param exportDirectory  path to the destination
+     * @param warnings         any warnings generated during this operation (e.g., source file
+     *                         outside of {@code ddf.home}) will be added to this collection
+     */
+    public void copyFileFromJavaPropertyValue(@NotNull Path propertyFilePath,
+            @NotNull String javaProperty, @NotNull Path exportDirectory,
+            @NotNull Collection<MigrationWarning> warnings) {
+        notNull(propertyFilePath, "Java properties file cannot be null");
+        Properties properties = readPropertiesFile(ddfHome.resolve(propertyFilePath));
+        String source = (String) properties.get(javaProperty);
+        notEmpty(source,
+                String.format("Source path property [%s] is invalid: [%s]", javaProperty, source));
+
+        Path sourcePath = Paths.get(source);
+
+        copy(sourcePath,
+                exportDirectory,
+                warnings,
+                () -> isSourceMigratable(sourcePath,
+                        () -> new PathMigrationWarning(propertyFilePath,
+                                javaProperty,
+                                sourcePath,
+                                "is absolute"),
+                        () -> new PathMigrationWarning(propertyFilePath,
+                                javaProperty,
+                                sourcePath,
+                                "contains a symbolic link"),
+                        () -> new PathMigrationWarning(propertyFilePath,
+                                javaProperty,
+                                sourcePath,
+                                String.format("is outside [%s]", ddfHome)),
+                        warnings));
+    }
+
+    private String getSystemProperty(String property) throws MigrationException {
+        String prop = System.getProperty(property);
+
+        if (StringUtils.isBlank(prop)) {
+            String message = String.format("System property %s is not set.", property);
+            LOGGER.error(message);
+            throw new MigrationException(message);
+        }
+
+        return prop;
+    }
+
+    private void copy(Path sourceFile, Path exportDirectory, Collection<MigrationWarning> warnings,
+            BooleanSupplier isSourceMigratable) {
+        notNull(sourceFile, "Source file cannot be null");
+        notNull(exportDirectory, "Destination cannot be null");
+        notNull(warnings, "Warning collection cannot be null");
+
```

Comment:
Do you think it is better to have the notNull() checks in the methods that have the @NotNull annotations?